### PR TITLE
OCPBUGS-87906: Fixed GetVCenter to handle removed FDs better

### PR DESCRIPTION
**Pull request by vr4manta**
OCPBUGS-87906

### Changes
- Enhanced GetVCenter logic to better handle finding vCenters for current node

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved vCenter selection for nodes with region/zone topology labels by consistently applying failure-domain matching.
  * Refined fallback behavior for single failure-domain and reduced configurations, including deterministic selection when applicable.
  * Added clearer error handling when vCenter resolution can’t be determined, and warning behavior when topology labels are missing.
* **Tests**
  * Expanded automated coverage for multi-vCenter/multi–failure-domain edge cases and regression scenarios after reducing to fewer failure domains.

### CODE REVIEW

File: pkg/check/util.go [hunk 1]
```diff
@@ -334,26 +340,47 @@ func GetVCenter(checkContext *CheckContext, node *v1.Node) (*VCenter, error) {
...
+			}
+
+			// Single FD cluster (or auto-generated FD): fall through to the first-FD fallback below
+			// so that the existing OCPBUGS-59319 behaviour is preserved.
+			klog.V(4).Infof("Node %s has region/zone labels but only one failure domain is configured; using it directly", node.Name)
+		}
+
+		// Node labels are absent, or we have exactly one FD and labels didn't match it (single-FD
+		// legacy cluster).  Fall back: return the first available vCenter/FD.
+		if len(region) == 0 || len(zone) == 0 {
```
Comment (gnufied) [inline pkg/check/util.go:366]: I think that before we fallback to returning first vCenter from FD configuration, we should also add an additional check here that returns error, if `region` and `zone` were not set and number of FDs was *greater* than 1, then there is some misconfiguration in the cluster and we should not blindly fallback to first vCenter.

File: pkg/check/util_test.go [hunk 2]
```diff
@@ -462,6 +462,97 @@ func TestGetVCenter(t *testing.T) {
 			}(),
 			server: "vcenter.test.openshift.com",
 		},
+		// ── Scenarios exercising the fix for the "FD reduced from 2→1" regression ──────────────
+		//
+		// Previously the matching path was gated on len(FailureDomains) > 1, so a node whose
```
Comment (gnufied) [inline pkg/check/util_test.go]: I am not sure how much this "Previously the matching" comment will make sense 2 years from now. The comments should be stand alone and make sense on their own.

File: pkg/check/util_test.go [hunk 3]
```diff
@@ -462,6 +462,97 @@ func TestGetVCenter(t *testing.T) {
...
+			infra:       testlib.InfrastructureWithFailureDomain(),
+			node: func() *v1.Node {
+				node := testlib.DefaultNodes()[0]
+				node.Labels = make(map[string]string)
+				node.Labels[v1.LabelFailureDomainBetaRegion] = "east"
+				node.Labels[v1.LabelFailureDomainBetaZone] = "east-1a"
+				return node
+			}(),
+			server: "dc0",
+		},
```
Comment (gnufied) [inline pkg/check/util_test.go:496]: So these 2 tests will IMO return identical results without *any* code change in `util.go`, because logic was not incorrect and we are not validating warning anyways in tests.  So, what are we even testing?

File: pkg/check/util.go (hunk 1 above)
Comment (vr4manta) [reply pkg/check/util.go:366]: sure, I can add this logic.

File: pkg/check/util_test.go (hunk 2 above)
Comment (vr4manta) [reply pkg/check/util_test.go]: Ok, i'll try to clean up w/ the intent.  Originally this block of code handled logic correctly until a change was made to add an additional check in the if statement. History will repeat itself if no documentation is left to guide dev through when new fix is required.

File: pkg/check/util_test.go (hunk 3 above)
Comment (vr4manta) [reply pkg/check/util_test.go:496]: AI suggested it and I am happy to always have more tests.   If you believe this is in error, then I will remove this test.

Comment (gnufied) [reply pkg/check/util_test.go:496]: >If you believe this is in error, then I will remove this test.

I do not know. Do you think these tests are reproducing the error customer had and catching the issue? Does the removing the change in util.go will cause these tests to fail?

Comment (vr4manta) [reply pkg/check/util_test.go:496]: Its using different labels that can be present on the nodes that we do use for the flow.  Results would end up the same, but the inputs are different labels.  I honestly do not know how many clusters will still have these way older labels on them.  If we should not see those anymore, then removing the test makes sense, else I'd say since its a valid input, putting a test in place to make sure its not broken as a good thing.

File: pkg/check/util.go [hunk 4]
```diff
@@ -334,26 +340,51 @@ func GetVCenter(checkContext *CheckContext, node *v1.Node) (*VCenter, error) {
...
+			// than one vCenter and the node's labels don't correspond to any configured FD (e.g.
+			// the FD was removed).  Return an error so the caller knows the node is unresolvable
+			// rather than silently returning the wrong vCenter.
+			if len(checkContext.PlatformSpec.FailureDomains) > 1 {
+				return nil, fmt.Errorf("unable to determine vcenter for node %s: no failure domain matched region=%s zone=%s", node.Name, region, zone)
+			}
+
+			// Single FD cluster (or auto-generated FD): fall through to the first-FD fallback below
+			// so that the existing OCPBUGS-59319 behaviour is preserved.
+			klog.V(4).Infof("Node %s has region/zone labels but only one failure domain is configured; using it directly", node.Name)
```
[the remark this answers is not in the document]
Comment (vr4manta) [reply pkg/check/util.go:361]: The logging is V(4) so our default logging will not have this; however, when we increase the logging, this is something I would need for troubleshooting issues.

**Comment by vr4manta** [comment]
/assign @gnufied 
Ready for next review

**Comment by vr4manta** [comment]
/verified by @vr4manta by manual tests and ci
